### PR TITLE
Mark skeleton for deprecation.

### LINKED ISSUE #4640: Mark conda-build skeleton as "Pending deprecation"
### What is the idea?

We eventually want to remove conda-skeleton from conda-build and instead have [Grayskull](https://github.com/conda-incubator/grayskull) be the de facto way for generating new recipes from existing packages.

### What should happen?

To help us achieve this goal, one of the first steps is to mark this feature as "pending deprecation" according to conda's official deprecation policy outlined in [CEP-9](https://github.com/conda-incubator/ceps/blob/main/cep-9.md). 

### Tasks

- [x] When conda-build users run `conda skeleton`, they need to be presented with a "pending deprecation" warning that will link them to the Grayskull project
- [ ] https://github.com/conda-incubator/ceps/pull/17 needs a vote
- [ ] Determine when this will receive the official "deprecation" status.
  - According to CEP-9, the "pending deprecation" status must last at least two regular releases. Do we want even more time for this?

### PR DESCRIPTION
**Pull request by jezdez**
### Description

This follows more or less the same pattern as #4645. Fixes #4640.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- ~[ ] Add / update necessary tests?~
- ~[ ] Add / update outdated documentation?~

### CODE REVIEW

File: conda_build/cli/main_skeleton.py [hunk 1]
```diff
@@ -46,6 +49,13 @@ def execute(args):
     parser, args = parse_args(args)
     config = Config(**args.__dict__)
 
+    deprecated.topic(
+        "3.25.0",
+        "4.0.0",
+        topic="`conda-build skeleton` and `conda skeleton`",
```
Comment (kenodegard) [inline conda_build/cli/main_skeleton.py:55]:
```suggestion
        topic="`conda skeleton` and `conda-skeleton`",
```

File: news/4640-use-grayskull [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* `conda-build skeleton` is pending deprecation. `conda-build` still provides the
+  `conda-build skeleton` a.k.a. `conda skeleton` CLI for now, but will remove
+  it eventually.
```
Comment (kenodegard) [inline news/4640-use-grayskull:13]:
```suggestion
* `conda skeleton` is pending deprecation. `conda-build` still provides
  `conda-skeleton` a.k.a. `conda skeleton` CLI for now, but will remove
  it eventually.
```

File: conda_build/cli/main_skeleton.py (hunk 1 above)
Comment (kenodegard) [inline conda_build/cli/main_skeleton.py:55]:
```suggestion
        topic="`conda skeleton` and `conda-skeleton`",
```

**Comment by dholth** [comment]
Seems pretty simple, straightforward.

File: news/4640-use-grayskull (hunk 2 above)
Comment (jezdez) [reply news/4640-use-grayskull]: D'oh!

Comment (dholth) [reply news/4640-use-grayskull]: At least the CLI plugin system is a single, less confusing way to have more CLI commands.

```
% ls conda-*
conda-build             conda-grayskull         conda-render
conda-convert           conda-greyskull         conda-server
conda-debug             conda-index             conda-skeleton
conda-develop           conda-inspect           conda-verify
conda-env               conda-metapackage
```

**Comment by jezdez** [comment]
pre-commit.ci autofix

File: news/4640-use-grayskull [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+* <news item>
+
+### Deprecations
+
+* `conda skeleton` is pending deprecation. `conda-build` still provides
+  `conda-skeleton` a.k.a. `conda skeleton` CLI for now, but will remove
+  it eventually.
+* Prefer the [standalone grayskull package](https://conda.github.io/grayskull/),
+  instead of `conda-build skeleton` or `conda skeleton`.
```
Comment (kenodegard) [inline news/4640-use-grayskull:15]:
```suggestion
  instead of `conda skeleton` or `conda-skeleton`.
```

**Review by travishathaway — DISMISSED** [review]
LGTM

**Review by beeankha — APPROVED** [review]

File: conda_build/cli/main_skeleton.py [hunk 4]
```diff
@@ -46,6 +49,13 @@ def execute(args):
     parser, args = parse_args(args)
     config = Config(**args.__dict__)
 
+    deprecated.topic(
+        "3.25.0",
```
Comment (kenodegard) [inline conda_build/cli/main_skeleton.py:53]:
```suggestion
        "3.26.0",
```

File: conda_build/skeletons/pypi.py [hunk 5]
```diff
@@ -253,6 +248,7 @@ def _formating_value(attribute_name, attribute_value):
     return " " + str(attribute_value) + "\n"
 
 
+@deprecated("3.25.6", "4.0.0", addendum="Use `grayskull` instead.")
```
Comment (kenodegard) [inline conda_build/skeletons/pypi.py:251]:
```suggestion
@deprecated("3.26.0", "4.0.0", addendum="Use `grayskull` instead.")
```

File: conda_build/skeletons/cpan.py [hunk 6]
```diff
@@ -390,7 +391,7 @@ def get_core_modules_for_this_perl_version(version, cache_dir):
     )
 
 
-# meta_cpan_url="http://api.metacpan.org",
+@deprecated("3.26.0", "4.0.0", addendum="Use grayskull instead.")
```
Comment (kenodegard) [inline conda_build/skeletons/cpan.py:394]:
```suggestion
@deprecated("3.26.0", "4.0.0", addendum="Use `grayskull` instead.")
```

File: conda_build/cli/main_skeleton.py [hunk 7]
```diff
@@ -46,6 +49,13 @@ def execute(args):
...
+        "4.0.0",
+        topic="`conda skeleton` and `conda-skeleton`",
+        addendum="Use the `grayskull` project instead.",
```
Comment (kenodegard) [inline conda_build/cli/main_skeleton.py:56]: Should we update all of these addendums to say something more informative:

```
Use the standalone `grayskull` project, see https://github.com/conda/grayskull.
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: conda_build/cli/main_skeleton.py (hunk 7 above)
Comment (kenodegard) [reply conda_build/cli/main_skeleton.py:56]: This is effectively what we have in the help blurb so I'm ok with not stressing over this and leaving the warnings as-is.

Comment (jezdez) [reply conda_build/cli/main_skeleton.py:56]: Yeah, let's keep it in the help

**Comment by dbast** [comment]
Are we sure we really want to deprecate the cran skeleton already?
* This afaik is still true: https://github.com/conda/grayskull/pull/423#issuecomment-1403798517
* The existing cran skeleton afaik was still used during the last Anaconda cran build out, as it can generate like almost 100% buildable recipes
* conda-forge (+ https://github.com/bgruening/conda_r_skeleton_helper maintained by the conda-forge/r group) still relies on it.

**Comment by jezdez** [comment]
> Are we sure we really want to deprecate the cran skeleton already?
> 
> * This afaik is still true: [[WIP]cran support grayskull#423 (comment)](https://github.com/conda/grayskull/pull/423#issuecomment-1403798517)
> * The existing cran skeleton afaik was still used during the last Anaconda cran build out, as it can generate like almost 100% buildable recipes
> * conda-forge (+ [bgruening/conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper) maintained by the conda-forge/r group) still relies on it.

For the record, this change **_does not deprecate skeleton_** but simply marks the code as **_pending deprecation_**, meaning that we intend to deprecate it in the future. We've set the time of deprecation to happen in the 4.0.0 release (currently).


> **Note**
> We've discussed adopting [CEP 8](https://github.com/conda-incubator/ceps/blob/main/cep-8.md) and [CEP 9](https://github.com/conda-incubator/ceps/blob/main/cep-9.md) for conda-build as well, so that the version number might change to the applicable version number. The same applies, pending deprecation means, it's far in the future and enough time to evolve the code and work on grayskull.

This change will provide both maintainers and users a predictable time window to solve the issues listed above by working with @marcelotrevisani and @bgruening (et al) to update grayskull. The alternative is an endless hand-wringing about how we can overcome the maintenance issues in skeleton.

If Anaconda is intending to use skeleton for the next cran build out, they may still do that a) until 4.0.0 is released (if they have to upgrade) or b) by using an older version of conda-build.

Of course, I hope by then we'll have enough Anaconda build engineers contribute actively to grayskull 😃

**Comment by kenodegard** [comment]
before we proceed with marking this for deprecation I'd like to see a `conda-skeleton` vs. `grayskull` comparison table and a promise to support the missing features in the near future

perhaps it's more appropriate to deprecate the functionality of `conda-skeleton` that `grayskull` already does but still keep the stuff that has not been implemented yet?

**Pull request closed without merging**